std/assembly/string.ts

```diff
@@ -80,11 +80,11 @@ import { Array } from "./array";
   }
 
   @operator("+") private static __concat(left: String, right: String): String {
-    return select<String>(left, changetype<String>("null"), left !== null).concat(right);
+    return left !== null ? left.concat(right) : right;
   }
 
   concat(other: String): String {
-    if (other === null) other = changetype<String>("null");
+    if (other === null) return this;
```

Comment:
Iirc this case is left over from the early days, just in case. Perhaps we can remove the line entirely nowadays?

Reply:
I'm not sure about this. In theory still may got situation when `other` is `null` with non-nullish type. For example when this string comes from host.

Reply:
another variant:
```ts
var a: string = "a";
var b: string | null = "b";
a + b
```

Reply:
Yeah, it is possible to construct such a situation with an export, yet I wonder whether guarding for invalid data from external sources at all the places is desirable as we also want stuff to be efficient internally. If I'm not mistaken we are not guarding for this anywhere else, just here, so unless we want to establish that convention everywhere in stdlib, the alternative is to not bother, even not here.

Reply:
I removed this checks